### ISSUE TITLE
[NEW] Send slot migration updates by push messages.

### ISSUE THREAD

**Issue by nihohit** [madolson/placeholderkv#1445]
**The problem/use-case that the feature addresses**

When slots are migrated between cluster nodes, the clients don't receive real-time updates on that. Instead, the clients rely on ASK and MOVED errors, and sometimes timer-based checks which incur a network & compute overhead.

**Description of the feature**

When a node changes its slot layout, it should send a push message to all RESP3 connections with its current slot layout, and which slots were added/removed, and if slots were moved to another node, to which node were the slots moved. This will allow clients to receive real-time updates, instead of relying on after-the-fact checks. Additionally, in the case of new nodes being added to a cluster, this allows the client to be updated about the new nodes without an expensive CLUSTER SLOTS/NODES/SHARDS call.

**Comment by ranshid** [comment]
@nihohit Great hearing form you!

Is this conceptually different than https://github.com/valkey-io/valkey/issues/57?
We have the [draft](https://github.com/valkey-io/valkey/pull/298) pending by @zuiderkwast that is meant to support this ask.
@zuiderkwast do you plan to progress on the draft or you would like to handover?

**Comment by nihohit** [comment]
I can't believe that I forgot my own proposal! 🙀
No, it's the same proposal, with the added idea to send to which nodes
slots were migrated, in order to save some work topology changes.

**Comment by ranshid** [comment]
ACK so I will close as duplicated

**Comment by ranshid** [comment]
Duplicate of https://github.com/valkey-io/valkey/issues/57
